securedrop/source.py

```diff
@@ -263,7 +262,7 @@ def submit():
         if entropy_avail >= 2400:
             async_genkey(g.sid, g.codename)
 
-    g.source.last_updated = datetime.utcnow()
+    g.source.last_updated = datetime.now().date()
```

Comment:
Yes I think this was incorrect,  c1587b3a99ba5b9ec7382aa25450b25950635c91 adds conversion to utc.

utcnow().date() is possible, it just changes *which* midnight is used. "touch -t "${date}0000" truncates to midnight from the server's local timezone, and datetime.now().date() is a roughly equivalent truncation in python.

If the journalist logs on at 9am and is in the same timezone as the server, then a submission with "less than a day ago" will have been sumitted on that same date, whereas "1 day ago" would have been submitted yesterday in their timezone.

This is a bit of a kludgy guess because of course the organisation could set the server timezone to something other than the journalist's time, but UTC is even more likely to be different.

At the end of the day it's hard to truncate the time in a delicate way, because useful display information is being discarded. A slightly more engineered approach would be to store the timezone ID that these truncations are occuring in *somewhere*, in case that becomes useful information later, but it wouldn't allow a more accurate display. Another option would be to allow the truncation timezone to be configured as something other than the server's local timezone in the UI.